mr/src/test/java/org/apache/iceberg/mr/hive/TestHiveIcebergStorageHandlerTimezone.java

```diff
@@ -0,0 +1,164 @@
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.mr.hive;
+
+import java.io.IOException;
+import java.text.DateFormat;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.util.Collection;
+import java.util.List;
+import java.util.TimeZone;
+import org.apache.hadoop.hive.serde2.io.DateWritable;
+import org.apache.hadoop.hive.serde2.io.TimestampWritable;
+import org.apache.iceberg.FileFormat;
+import org.apache.iceberg.Schema;
+import org.apache.iceberg.common.DynFields;
+import org.apache.iceberg.data.Record;
+import org.apache.iceberg.mr.TestHelper;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableList;
+import org.apache.iceberg.types.Types;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import static org.apache.iceberg.types.Types.NestedField.optional;
+import static org.junit.runners.Parameterized.Parameter;
+import static org.junit.runners.Parameterized.Parameters;
+
+@RunWith(Parameterized.class)
+public class TestHiveIcebergStorageHandlerTimezone {
+  DynFields.StaticField<ThreadLocal<DateFormat>> dateFormat = DynFields.builder()
+      .hiddenImpl(TimestampWritable.class, "threadLocalDateFormat")
+      .defaultAlwaysNull()
+      .buildStatic();
+
+  DynFields.StaticField<ThreadLocal<TimeZone>> localTimeZone = DynFields.builder()
+      .hiddenImpl(DateWritable.class, "LOCAL_TIMEZONE")
+      .defaultAlwaysNull()
+      .buildStatic();
+
+  @Parameters(name = "timezone={0}")
+  public static Collection<Object[]> parameters() {
+    return ImmutableList.of(
+        new String[] {"America/New_York"},
+        new String[] {"Asia/Kolkata"},
+        new String[] {"UTC/Greenwich"}
+    );
+  }
+
+  private TestHiveShell shell;
+
+  private TestTables testTables;
+
+  @Parameter(0)
+  public String timezoneString;
+
+  @Rule
+  public TemporaryFolder temp = new TemporaryFolder();
+
+  @Before
+  public void before() throws IOException, IllegalAccessException {
+    TimeZone.setDefault(TimeZone.getTimeZone(timezoneString));
+
+    // Magic to clean cached date format for Hive where the default timezone is used in the cached object
+    ThreadLocal<DateFormat> dynamicDateFormat = dateFormat.get();
+    if (dynamicDateFormat != null) {
+      dynamicDateFormat.remove();
+    }
+
+    // Magic to clean cached local timezone for Hive where the default timezone is stored in the cached object
+    ThreadLocal<TimeZone> dynamicTimeZone = localTimeZone.get();
+    if (dynamicTimeZone != null) {
+      dynamicTimeZone.remove();
+    }
+
+    this.shell = HiveIcebergStorageHandlerTestUtils.shell();
```

Comment:
Shouldn't we create a shell per test suite (w/ `BeforeClass`) like in the other test classes?

Reply:
Can we use parameterized test in this case? We need to set the default timezone to the one defined by the parameter.

Reply:
Good question. Not sure if calling `TimeZone.setDefault(TimeZone.getTimeZone(timezoneString));` will take effect if the shell/HS2 has been already started, but could be worth a try

Reply:
If not, we can just add a comment and leave it as it is (create a new shell for each test method)

Reply:
Ok.. so after the "magic" it is working now